### ISSUE TITLE
Producing gibberish

### ISSUE THREAD

**Issue by sair770** [UltimateHackers/Decodify#14]
![screenshot from 2018-04-15 19-22-31](https://user-images.githubusercontent.com/13485983/38779219-0b4cb3a0-40e3-11e8-80e9-10c1f5e2a8c6.png)
This was occuring because the regex search identifies "637a42745a444e32" as both hex and base64 encoded string. This can be avoided if lines like these
`   decode(base, 'url')`
`   decode(string, 'none')`

are replaced with 
`   decode(base, 'url')`
`    decode(string, 'none')`

**Comment by UltimateHackers** [comment]
Decodify decodes a string for as many as encodings it matches.
This feature was later included after some people reported false positives.
Btw the lines you have suggested to replace are same as the lines you have suggested to be replaced by.

**Comment by sair770** [comment]
Sorry for that.edited the comment, I also submitted a pr.

Yeah I kind of understand it does decode every encoding it matches but not every regex it matches is a correct encoding. You can check the example from readme.md it doesnt produce exact result.

**Comment by UltimateHackers** [comment]
Let say, encoding **A** uses the chars `abc` and encoding **B** uses the chars `abcd`.
Now you have a string `bcbabb`, what encoding does it use? **A** or **B**??
We can't decide right? That's the point.

**Comment by sair770** [comment]
I might have not worded my sentence correctly yeah you are correct at what you saying what the actual issue is that when I run a string it should first complete all the possible encodings it detected first then try for rest encodings
like lets take k=acbbca as encoded string but when i first detect it as encoding A then we decode k by A lets assume we get k' then we have to work on k' until we completely exhaust our options instead what is happening here is we are working till k' and searching if k matches any other encoding in middle of the process.

![image](https://user-images.githubusercontent.com/13485983/38973353-00cda99c-43c3-11e8-9a06-3daff2a0dd69.png)

So the problem i encountered is because of that and you can clearly see in the screenshot above as I have used the example from readme.md

![image](https://user-images.githubusercontent.com/13485983/38973423-532a0424-43c3-11e8-8e5d-0ee87c80ae7c.png)
This is screenshot after I made the changes i put in the issue.

**Comment by sair770** [comment]
if you wonder why it is happening is .
first when i try to identify NjM3YTQyNzQ1YTQ0NGUzMg== i get base64 and pass it base64 function where it decodes to  637a42745a444e32 

next step is you try to identify if it(NjM3YTQyNzQ1YTQ0NGUzMg==) matches any other encoding then you go try to identify 637a42745a444e32 which at first correctly identifies as hex and decodes as czBtZDN2 now again as per the code the program tries to identify any other match for the  base(637a42745a444e32
637a42745a444e32) so this returns base64 which is wrong but it goes to decode it wrongly returns gibberish and that is again identified and hex when decoded it returns czBtZDN2 which is already present in the decoded list so it doesnt really go to actual decoding of the czBtZDN2 and return s0md3v it broke down at misidentifying 637a42745a444e32 as base64 and breaking the flow to reidentify the string 

**TL;DR**
what i want to suggest is we should have way to check if the decoded string is ascii chars or something(which am not exactly sure) and and do a roll back if it is not and first decode the string in its entirety and then go for checking if the original string and intermediate strings match any other encodings
